src/variable.rs

```diff
@@ -35,6 +36,10 @@ impl Variable {
     pub fn vartype(&self) -> nc_type {
         self.vartype
     }
+    /// Get current length of the variable
+    pub fn len(&self) -> usize {
+        self.dimensions.iter().map(|d| d.len()).product()
```

Comment:
This product risks silently wrapping around type limits (in release compiles), if that's a concern. It at least means that this variable cannot be trusted in all unsafe code.

Reply:
This one in particular can't overflow, as the lengths of the dimensions are backed by the netcdf library. Should probably introduce a check for slice_len everywhere else though.

Reply:
I was looking at the product inside `fn values` and it looks like it has this problem. But if there is an invariant ensuring we can never have a product this big, then it's not a problem.

Reply:
Using saturating multiplication and erroring on hitting max bound now